image_compare/test_spec.js

```diff
@@ -0,0 +1,207 @@
+/* global suite, setup, test, XMLHttpRequest, p5 */
+
+suite('Image Compare', function() {
+  
+  var showFailures = true;
+  var showAll = false;
+  
+  function showRef(refData) {
+    var canvas = document.createElement('canvas');
+    canvas.width = 100;
+    canvas.height = 100;
+    var ctx = canvas.getContext('2d');
+    var imgData = ctx.getImageData(0,0,100,100);
+    for (var i = 0; i < refData.length; i++) {
+      imgData.data[i] = refData[i];
+    }
+    ctx.putImageData(imgData, 0, 0);
+    document.body.appendChild(canvas);
+  }
+  
+  function calcSameness(refData, actData) {
+    var n = 0;
+    
+    for (var i = 0; i < actData.length; i++) {
+      n += Math.abs(actData[i] - refData[i]);
+    }
+
+    return n / (256 * 4 * actData.length);
+  }
+  
+  function getRefData(code) {
+    var line = code.split('\n')[0];
+    var regex = /\/\/\[[0-9]+\,[0-9]+\]/;
+    var matches = regex.exec(line);
+
+    var len = matches[0].length;
+    return line.substring(len).split(',').map(function (val) {
+      return parseInt(val);
+    });
```

Comment:
The first line of each .pde file contains a comment that looks something like this:
```//[100,100]204,204,204,255,204,204,204,255,...```
The first line contains the reference image data.